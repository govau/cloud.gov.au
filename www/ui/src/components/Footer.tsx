import * as React from "react";
import styled, { css } from "../styled-components";
import { Flex, Box } from "grid-styled";

import * as coaLogo from "./coa.svg";
import * as dtaLogo from "./dta.svg";
import Container from "./Container";
import Link from "./Link";

const COALogo = styled.img`
  height: 5rem;
`;

const DTALogo = styled.img`
  height: 3rem;
`;

interface LogoSeparatorProps {
  className?: string;
}

const UnstyledLogoSeparator: React.StatelessComponent<LogoSeparatorProps> = ({
  className
}) => (
  <React.Fragment>
    <Box width={[1 / 16]} className={className} />
    <Box width={[1 / 16]} />
  </React.Fragment>
);

const LogoSeparator = styled(UnstyledLogoSeparator)`
  height: 3rem;
  ${({ theme }) =>
    css`
      border-right: 2px solid ${theme.footerLogoDividerColor};
    `};
`;

const Copyright = styled.p`
  ${({ theme }) =>
    css`
      color: ${theme.footerTextColor};
    `};
`;

const StyledFooter = styled.footer`
  ${({ theme }) =>
    css`
      background-color: ${theme.footerColor};
    `};
  color: #ffffff;
`;

interface Props {}

const Footer: React.StatelessComponent<Props> = () => (
  <StyledFooter>
    <Container>
      <Flex wrap={true} pt={3} pb={3} align="center">
        <Box w={[1, 1 / 2, 1 / 3]}>
          <Flex wrap={true} pt={3} pb={2} align="center">
            <Box>
              <COALogo src={coaLogo} alt="" />
            </Box>
            <LogoSeparator />
            <Box>
              <a href="https://www.dta.gov.au/">
                <DTALogo src={dtaLogo} alt="DTA" />
              </a>
            </Box>
          </Flex>
        </Box>
        <Box w={[1, 1 / 2, 2 / 3]}>
          <p>
            <Link href="http://status.cloud.gov.au/">Status</Link>
          </p>
          <p>
            <Link href="https://github.com/govau/cloud.gov.au">
              cloud.gov.au on GitHub
            </Link>
          </p>
          <p>
            Have a question about cloud.gov.au? Contact DTA:{" "}
            <Link href="mailto:support@cloud.gov.au">support@cloud.gov.au</Link>
          </p>
          <Copyright>
            Â© Commonwealth of Australia. With the exception of the Commonwealth
            Coat of Arms and where otherwise noted, this work is licensed under
            the CC BY 4.0 license.
          </Copyright>
        </Box>
      </Flex>
    </Container>
  </StyledFooter>
);

export default Footer;
