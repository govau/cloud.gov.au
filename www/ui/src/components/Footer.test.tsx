import * as React from "react";
import * as renderer from "react-test-renderer";

import Footer from "./Footer";

it("renders without crashing", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
