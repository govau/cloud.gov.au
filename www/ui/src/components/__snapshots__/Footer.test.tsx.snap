// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<footer
  className="sc-iwsKbI XbfZB"
>
  <div
    className="sc-bdVaJa liJgVB sc-bdVaJa bNqJOV"
    is={null}
  >
    <div
      className="sc-htpNat fzUmoZ sc-bdVaJa NcCKj"
      is={null}
    >
      <div
        className="sc-bdVaJa dbOKEe"
        is={null}
      >
        <div
          className="sc-htpNat fzUmoZ sc-bdVaJa kQtcea"
          is={null}
        >
          <div
            className="sc-bdVaJa iHZvIS"
            is={null}
          >
            <img
              alt=""
              className="sc-bZQynM HqUwQ"
              src={
                Object {
                  "0": "c",
                  "1": "o",
                  "2": "a",
                  "3": ".",
                  "4": "s",
                  "5": "v",
                  "6": "g",
                  "default": "coa.svg",
                }
              }
            />
          </div>
          <div
            className="sc-htoDjs bWdAUX sc-bdVaJa jGvYNb"
            is={null}
          />
          <div
            className="sc-bdVaJa jGvYNb"
            is={null}
          />
          <div
            className="sc-bdVaJa iHZvIS"
            is={null}
          >
            <a
              href="https://www.dta.gov.au/"
            >
              <img
                alt="DTA"
                className="sc-gzVnrw cJnkpP"
                src={
                  Object {
                    "0": "d",
                    "1": "t",
                    "2": "a",
                    "3": ".",
                    "4": "s",
                    "5": "v",
                    "6": "g",
                    "default": "dta.svg",
                  }
                }
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="sc-bdVaJa clWLZM"
        is={null}
      >
        <ul
          className="sc-bxivhb ddELEZ"
        >
          <li
            className="sc-ifAKCX jMGxLf"
          >
            <a
              className="sc-EHOje jvzENE sc-bwzfXH gRAcqg"
              href="https://console.system.b.cld.gov.au/"
            >
              Production console
            </a>
          </li>
          <li
            className="sc-ifAKCX jMGxLf"
          >
            <a
              className="sc-EHOje jvzENE sc-bwzfXH gRAcqg"
              href="https://console.system.y.cld.gov.au/"
            >
              Staging console
            </a>
          </li>
          <li
            className="sc-ifAKCX jMGxLf"
          >
            <a
              className="sc-EHOje jvzENE sc-bwzfXH gRAcqg"
              href="http://status.cloud.gov.au/"
            >
              Status
            </a>
          </li>
        </ul>
        <p>
          <a
            className="sc-bwzfXH gRAcqg"
            href="https://github.com/govau/cloud.gov.au"
          >
            cloud.gov.au on GitHub
          </a>
        </p>
        <p>
          Have a question about cloud.gov.au? Contact DTA:
           
          <a
            className="sc-bwzfXH gRAcqg"
            href="mailto:support@cloud.gov.au?subject=Enquiry%20about%20cloud.gov.au&body=%0A%0A---%0ASource%3A%20cloud.gov.au%20footer"
          >
            support@cloud.gov.au
          </a>
        </p>
        <p
          className="sc-dnqmqq eyiWaz"
        >
          Â© Commonwealth of Australia. With the exception of the Commonwealth Coat of Arms and where otherwise noted, this work is licensed under the CC BY 4.0 license.
        </p>
      </div>
    </div>
  </div>
</footer>
`;
